src/dappsExplorer/search/DAppsExplorerScreenSearch.tsx

```diff
@@ -0,0 +1,327 @@
+                      <Text style={styles.sectionTitle}>
+                        {t('dappsScreen.favoriteDapps').toLocaleUpperCase(language ?? 'en-US')}
+                      </Text>
+                    )}
+                    <FavoriteDappsSection
+                      onPressDapp={onSelectDapp}
+                      searchTerm={searchTerm}
+                      setFavoriteResultsEmpty={setFavoriteResultsEmpty}
+                    />
+                    {/* If all dapp section isn't empty or favoriteResults isn't empty display add section header */}
+                    {(!allResultEmpty || !favoriteResultsEmpty) && (
+                      <Text style={styles.sectionTitle}>
+                        {t('dappsScreen.allDapps').toLocaleUpperCase(language ?? 'en-US')}
+                      </Text>
+                    )}
+                  </>
+                )}
+              </>
+            }
+            style={styles.sectionList}
+            contentContainerStyle={[
+              styles.sectionListContentContainer,
+              { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) },
+            ]}
+            // Workaround iOS setting an incorrect automatic inset at the top
+            scrollIndicatorInsets={{ top: 0.01 }}
+            scrollEventThrottle={16}
+            onScroll={onScroll}
+            // @ts-expect-error Type 'SectionData[]' is not assignable to type 'readonly SectionListData<DappV2, SectionData>[] | AnimatedNode<readonly SectionListData<DappV2, SectionData>[]>'
+            sections={allSectionResults}
+            renderItem={({ item: dapp }) => (
+              <DappCard
+                dapp={dapp}
+                section={DappSection.All}
+                onPressDapp={onSelectDapp}
+                onFavoriteDapp={onFavoriteDapp}
+              />
+            )}
+            keyExtractor={(dapp: DappV1 | DappV2) => dapp.id}
+            stickySectionHeadersEnabled={false}
+            testID="DAppsExplorerScreenSearch/DappsList"
+            ListEmptyComponent={emptyListComponent}
+            ListFooterComponentStyle={styles.listFooterComponent}
+            keyboardShouldPersistTaps="always"
+            keyboardDismissMode="on-drag"
+          />
+        )}
+      </>
+      {DappFavoritedToast}
+      {DappInfoBottomSheet}
+    </SafeAreaView>
+  )
+}
+
+function DescriptionView({ message, title }: { message: string; title: string }) {
+  return (
+    <View style={styles.descriptionView}>
+      <Text style={styles.pageHeaderText}>{title}</Text>
+      <Text style={styles.pageHeaderSubText}>{message}</Text>
+    </View>
+  )
+}
+
+function parseResultsIntoAll(
+  dappList: (DappV1WithCategoryName | DappV2WithCategoryNames)[],
+  searchTerm: string,
+  favoriteDappsById: string[]
+) {
+  // If there is no search query, return the non favorite dapps in the all section
+  const nonFavoriteDapps = dappList.filter(
+    (dapp: DappV1 | DappV2) => !favoriteDappsById.includes(dapp.id)
```

Comment:
Can we do it without the explicit type?
```suggestion
    (dapp) => !favoriteDappsById.includes(dapp.id)
```